File: frontend/src/types/workflow.ts
/**
 * Type de nœud dans le workflow
 */
export type WorkflowNode = {
    id: string;
    tool_id: string; // Changed from 'tool' to match backend Pydantic model
    label: string;   // Added label
    position: { x: number; y: number }; // Added position
    data: Record<string, any>; // Changed from 'params' to 'data'
};

/**
 * Arête/lien entre deux nœuds
 */
export type WorkflowEdge = {
    from: string;
    to: string;
};

/**
 * Workflow complet
 */
export type Workflow = {
    id: string;
    sport: string;
    query?: string;
    nodes: WorkflowNode[];
    edges: WorkflowEdge[];
};

/**
 * Types de sports supportés
 */
export type SportType = "football" | "basketball" | "tennis" | "rugby";

File: backend/app/models/workflow.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional

class ToolModel(BaseModel):
    id: str
    name: str
    description: str
    supported_sports: List[str]

class WorkflowNode(BaseModel):
    id: str
    tool_id: str
    label: str
    position: Dict[str, float] = Field(default_factory=lambda: {"x": 0, "y": 0})
    data: Dict[str, Any] = Field(default_factory=dict)

class WorkflowEdge(BaseModel):
    id: str
    source: str
    target: str

class Workflow(BaseModel):
    id: str
    sport: str
    query: str
    nodes: List[WorkflowNode]
    edges: List[WorkflowEdge]

class WorkflowCreate(BaseModel):
    sport: str
    query: str

class WorkflowExecute(BaseModel):
    workflow: Workflow
